### PR TITLE
Fix UsageResourceTest.testListApplicationUsages

### PR DESCRIPTION
**Pull request by aledsage**
Context: this test is checking that the REST api for application-usage will report usage in the specified time range. If we ask for usage between `Wed Mar 18 11:45:25 GMT 2015` and `Wed Mar 18 11:45:25 GMT 2015` then we expect to be told about all apps that were in use during that single second (and not to be told about the usage of those apps outside of that specific second).

- Was failing occasionally if the requested start/end date rounded down to a different value than the postStart date.
- I suspect 1 in 1000 chance, because depended if afterPostStart was in a different second from postStart - it was adding 1ms to it. I suspect the original intent was to add 1 second.

### CODE REVIEW

**Comment by neykov** [comment]
I was the last one who took a stab at fixing this test so can provide some context.
The 1ms delay is on purpose due to the check in `UsageResource`
```
            if (eventStartDate.compareTo(endDate) > 0 || eventEndDate.compareTo(startDate) < 0) {
                continue;
            }
```
We have to make sure that we are requesting times at least 1ms *after* `eventStartDate` (thus the +1ms). The `waitForFuture(postStart+1ms)` is because `eventEndDate=new Date()`, to make sure that the requested time is not in the future. The comparison is done on the full values, they are cropped at serialization time only, so no need to add 1s.

Your other half of the fix is correct though, the test should assert against `afterPostStart` due to
```
            if (eventStartDate.compareTo(startDate) < 0) {
                eventStartDate = startDate;
            }
            if (eventEndDate.compareTo(endDate) > 0) {
                eventEndDate = endDate;
            }
```

I can see another potential problem. It could (quite unlikely) happen that `afterPostStart` is equal to `eventEndDate = new Date()` Due to the `eventEndDate.compareTo(startDate) < 0` the test will fail. To make sure this doesn't happen I'd suggest changing the original test to
```
long afterPostStart = postStart.getTime()+1;
waitForFuture(afterPostStart+1);
```
so that we are completely sure that `afterPostStart` is at least 1ms after the entity start time and at least 1ms before `new Date()`.

**Comment by aledsage** [comment]
Thanks @neykov - I've incorporated your suggestions.

Not sure I entirely follow. We're using `.compareTo(...) > 0` and `.compareTo(...) < 0`, so they should cope fine with it being equal. It should always return us event details about `startDate` to `endDate` (i.e. `afterPostStart` to `afterPostStart`, because that is what we're now passing in as the time range).

I've fine with the test taking an extra 1ms to be cautious though.

File: usage/rest-server/src/test/java/brooklyn/rest/resources/UsageResourceTest.java [hunk 1]
```diff
@@ -426,6 +432,10 @@ public void release(SshMachineLocation machine) {
         }
     }
 
+    private void waitForFuture(Date futureTime) throws InterruptedException {
```
Comment (neykov) [inline usage/rest-server/src/test/java/brooklyn/rest/resources/UsageResourceTest.java]: Looks like this is not used any more?

**Comment by neykov** [comment]
You are right, the comparison is inclusive. LGTM with or without the extra ms.

**Pull request merged**